core/base/constrainedGradientDescent/ConstrainedGradientDescent.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
```

Comment:
add doxygen documentation.
at least the doxygen preambule with a proper description of the class (see for instance `core/base/abstractTriangulation/AbstractTriangulation.h`).